integration/peers_bootstrap_none_available_test.go

```diff
@@ -0,0 +1,127 @@
+// +build integration
+
+// Copyright (c) 2018index Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package integration
+
+import (
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/m3db/m3cluster/services"
+	"github.com/m3db/m3cluster/shard"
+	"github.com/m3db/m3db/client"
+	"github.com/m3db/m3db/retention"
+	"github.com/m3db/m3db/sharding"
+	"github.com/m3db/m3db/storage/namespace"
+	"github.com/m3db/m3db/topology"
+	"github.com/m3db/m3db/topology/testutil"
+	xlog "github.com/m3db/m3x/log"
+
+	"github.com/stretchr/testify/require"
+)
+
+// TestPeersBootstrapNoneAvailable makes sure that a cluster with the peer bootstrapper enabled can still turn on
+// if none of the nodes peers shard states are available or leaving.
+func TestPeersBootstrapNoneAvailable(t *testing.T) {
```

Comment:
Nice, did we confirm whether or not a new cluster starts with INITIALIZING shards or AVAILABLE shards?

Reply:
Damn, think they start as available.  We may have to make a change to m3cluster:

Reply:
Yeah can probably still land this, and then make that change and update M3DB. The comment in that file seems like its more complicated than just changing the default value, will it affect our client as well?